model/src/main/java/org/projectnessie/api/params/NamespaceParams.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.api.params;
+
+import java.util.Objects;
+import java.util.StringJoiner;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import javax.ws.rs.PathParam;
+import org.eclipse.microprofile.openapi.annotations.media.ExampleObject;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameter;
+import org.immutables.builder.Builder;
+import org.projectnessie.model.Namespace;
+import org.projectnessie.model.Validation;
+
+public class NamespaceParams {
+
+  @Parameter(
+      description = "name of ref to fetch",
+      examples = {@ExampleObject(ref = "ref")})
+  @PathParam("ref")
+  @NotNull
+  @Pattern(regexp = Validation.REF_NAME_REGEX, message = Validation.REF_NAME_MESSAGE)
+  private String refName;
```

Comment:
Requires a `hashOnRef` as well - think: `DETATCHED`

Reply:
you're right, added